packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -337,6 +353,81 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
     this.setState({ menuProps: currentMenuProps ? null : menuProps });
   }
 
+  @autobind
```

Comment:
the caller is already in the context of the class so this should be pointing to what you want already.